draft-wussler-openpgp-pqc.md

```diff
@@ -976,24 +976,28 @@ scheme is as follows:
 
 ### Public-Key Encrypted Session Key Packets (Tag 1) {#ecc-kyber-pkesk}
 
-The composite Kyber algorithms MUST be used only with v6 PKESK, as defined in
-[I-D.ietf-openpgp-crypto-refresh] Section 5.1.2.
+The algorithm-specific fields consists of:
 
-The algorithm-specific v6 PKESK parameters consists of:
+ - The one-octet algorithm identifier, if it was passed (in the case of a v3 PKESK packet).
```

Comment:
I would not allow v3 PKESK with PQC. This adds IMO a security problem too: the alg ID in the v3 PKESK is inside the wrapped session key. In v6 PKESK, it is in plaintext in the encrypted data packet, but it is used again in an HKDF to derive the actual encryption key.
In this mix we open up to cross-algorithm attacks

Reply:
I discussed this with Daniel and the proposed changes to the X25519/X448 algos for the refresh.
I am fine with putting the algorithm unencrypted as long as we limit the usable symmetric algorithms to AES (7, 8, 9) only. Since 7 is a must anyway, we don't limit interoperability.

This guarantees that the key size is enough to determine which algorithm ID we're using. We can still put it there explicit, but it effectively prevents cross-algorithm attacks by limiting the algorithms.

Reply:
For the record: in the meantime we agreed that we will allow v3 PKESK for the use with PQC encryption: https://github.com/openpgp-pqc/draft-openpgp-pqc/issues/35#issuecomment-1576596480-permalink